### ISSUE TITLE
KeyError: 'p1' in cuckoo/processing/platform/linux.py, line 93

### ISSUE THREAD

**Issue by zentavr** [cuckoosandbox/cuckoo#2264]
#### My issue is:
I'm trying to build a sandbox with linuxes guest and found a couple of samples in the internet.
When i upload them via web page, I see this traceback:
```
Failed to run the processing module "BehaviorAnalysis" for task #296: 
Traceback (most recent call last):
    File "/opt/.cuckoovenv/local/lib/python2.7/site-packages/cuckoo/core/plugins.py", line 240, in process 
        data = current.run() 
    File "/opt/.cuckoovenv/local/lib/python2.7/site-packages/cuckoo/processing/behavior.py", line 351, in run 
        for event in handler.parse(path): 
    File "/opt/.cuckoovenv/local/lib/python2.7/site-packages/cuckoo/processing/platform/linux.py", line 76, in parse 
        self.post_hook(syscall)
    File "/opt/.cuckoovenv/local/lib/python2.7/site-packages/cuckoo/processing/platform/linux.py", line 93, in post_hook 
        pid["command_line"] = " ".join(syscall["arguments"]["p1"])
KeyError: 'p1'
```

I added some more logs and `syscall` really does not have `p1` key.
```
{'api': 'execve',
 'arguments': {'p0': '"/tmp/4654d78e184123e0dbb1d52baf04354c964c88d9450fd61308a2b7101c269b56_sample.elf" '},
 'instruction_pointer': '804e562',
 'pid': 1178,
 'process_name': '4654d78e184123e',
 'raw': 'Wed May  9 20:59:43 2018.139592 4654d78e184123e@804e562[1178] execve("/tmp/4654d78e184123e0dbb1d52baf04354c964c88d9450fd61308a2b7101c269b56_sample.elf" )\n',
 'return_value': '',
 'status': '',
 'time': datetime.datetime(2018, 5, 9, 20, 59, 43, 139592),
 'type': 'apicall'}
```

##### My Cuckoo version and operating system are:
`Cuckoo==2.0.4.4` on `Ubuntu 16.04.3 LTS`

**Comment by zentavr** [comment]
@doomedraven : i am trying to do an upgrade and have another error. I used vurtualenv with my previous setup (2.0.4.4) and created a new one for 2.0.5.3. when i do `cuckoo import` - I have this:
```
(.cuckoovenv2.0.5) cuckoo@cuckoo:/opt/cuckoo-ti4$ cuckoo import ../cuckoo-ti4.old/
You are importing an existing Cuckoo setup. Please
understand that, depending on the mode taken, if
you remove the old Cuckoo setup after this import
you may still lose ALL of your data!

Additionally, database migrations will be performed
in-place*. You won't be able to use your old Cuckoo
setup anymore afterwards! However, we'll provide
you with the option to create a SQL backup beforehand.

TL;DR Cleaning the old setup after the import may
corrupt your new setup: its SQL, MongoDB, and
ElasticSearch database may be dropped and, in 'symlink'
mode, the analyses removed.

*: Except for sqlite3 databases in combination with
   the import 'copy' approach.

... I've read the above and understand the consequences [y/N]: y
Traceback (most recent call last):
  File "/opt/.cuckoovenv2.0.5/bin/cuckoo", line 11, in <module>
    sys.exit(main())
  File "/opt/.cuckoovenv2.0.5/local/lib/python2.7/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/opt/.cuckoovenv2.0.5/local/lib/python2.7/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/opt/.cuckoovenv2.0.5/local/lib/python2.7/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/.cuckoovenv2.0.5/local/lib/python2.7/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/.cuckoovenv2.0.5/local/lib/python2.7/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/opt/.cuckoovenv2.0.5/local/lib/python2.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/opt/.cuckoovenv2.0.5/local/lib/python2.7/site-packages/cuckoo/main.py", line 659, in import_
    import_cuckoo(ctx.parent.user, mode, path)
  File "/opt/.cuckoovenv2.0.5/local/lib/python2.7/site-packages/cuckoo/apps/import_.py", line 129, in import_cuckoo
    "The path that you specified is not a proper Cuckoo setup. Please "
cuckoo.common.exceptions.CuckooOperationalError: The path that you specified is not a proper Cuckoo setup. Please point the path to the root of your older Cuckoo setup, i.e., to the directory containing the cuckoo.py script!
```

How can I fix that?

**Comment by doomedraven** [comment]
why do you need import ? CWD should be fine, i had updated without import command

**Comment by zentavr** [comment]
@doomedraven, I used https://cuckoo.sh/docs/installation/upgrade.html document in order to go.

**Comment by zentavr** [comment]
The function which performs the test is
```python
def identify(dirpath):
    filepath = os.path.join(dirpath, "lib", "cuckoo", "common", "constants.py")
    if os.path.exists(filepath):
        for line in open(filepath, "rb"):
            if line.startswith("CUCKOO_VERSION"):
                return line.split('"')[1]
```

I checked **lib/cuckoo/common/constants.py** and there is **CUCKOO_VERSION** only in version 1.2. Seems like this got removed since 2.x

**Comment by zentavr** [comment]
@doomedraven : the same error happens with 2.0.5.3 release.
**Analyzer Log** is:
```
2018-05-10 17:01:09,001 [root] DEBUG: Starting analyzer from: /tmpJ2vxFe
2018-05-10 17:01:09,001 [root] DEBUG: Storing results at: /tmp/YhcDKOVlNi
2018-05-10 17:01:09,129 [modules.auxiliary.stap] INFO: STAP aux module startup took 0.11 seconds
2018-05-10 17:01:09,129 [root] DEBUG: Started auxiliary module STAP
2018-05-10 17:01:09,130 [root] INFO: Added new process to list with pid: 1177
2018-05-10 17:01:10,133 [root] INFO: Process with pid 1177 has terminated
2018-05-10 17:01:10,133 [root] INFO: Process list is empty, terminating analysis.
2018-05-10 17:01:11,137 [modules.auxiliary.stap] DEBUG: stap subprocess retval None
2018-05-10 17:01:11,146 [root] INFO: Analysis completed.
```

**Cuckoo Log** is:
```
2018-05-10 17:01:09,152 [cuckoo.core.scheduler] INFO: Task #305: acquired machine debian8_x86_01 (label=debian8_x86_01)
2018-05-10 17:01:09,161 [cuckoo.auxiliary.sniffer] INFO: Started sniffer with PID 4130 (interface=vboxnet0, host=192.168.56.131)
2018-05-10 17:01:09,161 [cuckoo.core.plugins] DEBUG: Started auxiliary module: Sniffer
2018-05-10 17:01:09,175 [cuckoo.machinery.virtualbox] DEBUG: Starting vm debian8_x86_01
2018-05-10 17:01:09,263 [cuckoo.machinery.virtualbox] DEBUG: Restoring virtual machine debian8_x86_01 to vminit
2018-05-10 17:01:11,362 [cuckoo.core.guest] INFO: Starting analysis on guest (id=debian8_x86_01, ip=192.168.56.131)
2018-05-10 17:01:12,365 [cuckoo.core.guest] DEBUG: debian8_x86_01: not ready yet
2018-05-10 17:01:13,369 [cuckoo.core.guest] DEBUG: debian8_x86_01: not ready yet
2018-05-10 17:01:14,373 [cuckoo.core.guest] DEBUG: debian8_x86_01: not ready yet
2018-05-10 17:01:14,431 [cuckoo.core.guest] DEBUG: debian8_x86_01: not ready yet
2018-05-10 17:01:16,435 [cuckoo.core.guest] DEBUG: debian8_x86_01: not ready yet
2018-05-10 17:01:17,324 [cuckoo.core.guest] INFO: Guest is running Cuckoo Agent 0.7 (id=debian8_x86_01, ip=192.168.56.131)
2018-05-10 17:01:17,331 [cuckoo.core.guest] DEBUG: Uploading analyzer to guest (id=debian8_x86_01, ip=192.168.56.131, monitor=latest, size=30538)
2018-05-10 17:01:17,392 [cuckoo.core.guest] DEBUG: debian8_x86_01: analysis still processing
2018-05-10 17:01:17,433 [cuckoo.core.resultserver] DEBUG: LogHandler for live analysis.log initialized.
2018-05-10 17:01:18,398 [cuckoo.core.guest] DEBUG: debian8_x86_01: analysis still processing
2018-05-10 17:01:19,404 [cuckoo.core.guest] DEBUG: debian8_x86_01: analysis still processing
2018-05-10 17:01:19,590 [cuckoo.core.resultserver] DEBUG: File upload request for logs/all.stap
2018-05-10 17:01:19,597 [cuckoo.core.resultserver] DEBUG: Uploaded file length: 28834
2018-05-10 17:01:20,409 [cuckoo.core.guest] INFO: debian8_x86_01: analysis completed successfully
2018-05-10 17:01:20,467 [cuckoo.core.plugins] DEBUG: Stopped auxiliary module: Sniffer
2018-05-10 17:01:20,467 [cuckoo.machinery.virtualbox] DEBUG: Stopping vm debian8_x86_01
2018-05-10 17:01:22,608 [cuckoo.core.scheduler] DEBUG: Released database task #305
2018-05-10 17:01:22,619 [cuckoo.core.scheduler] INFO: Task #305: analysis procedure completed
```

**Comment by doomedraven** [comment]
```
2018-05-10 17:01:09,130 [root] INFO: Added new process to list with pid: 1177
2018-05-10 17:01:10,133 [root] INFO: Process with pid 1177 has terminated
```

see timing it was executed for 1 sec only

**Comment by zentavr** [comment]
@doomedraven : ok, maybe it was exacuted, but I can see the score only 0.4/10 and the error with the traceback I provided at the beginning.

I found that sample on public https://linux.huntingmalware.com/analysis/13318/summary/ and there (cuckoo 1337) the test provides 4.0/10 points with no errors.

**Comment by zentavr** [comment]
@doomedraven : Could it be because my debian guest OS does not have kernel's -dbgsym package installed?

**Comment by zentavr** [comment]
...while testing I tried to install Ubuntu 17.04. The problem is that distro is not supported anymore and seems like I cannot find and install kernel's -dbgsym package. Here in #1375 you recommend to use Ubuntu 17.04 as a guest.

I wonder if I can:
a) Use 17.10 or 18.04 which are supported
b) Use a kernel from different distro

**Comment by doomedraven** [comment]
@zentavr  https://linux.huntingmalware.com is my linux cuckoo, you always can try that

**Comment by doomedraven** [comment]
about the score, is bcz you need write your own signatures

**Comment by jbremer** [comment]
Closing issue, seems resolved.
